torch_audiomentations/augmentations/spliceout.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from turtle import forward
+import torch
+from typing import Optional
+from torch import Tensor
+from torch.nn.functional import pad
+
+from ..core.transforms_interface import BaseWaveformTransform
+from ..utils.dsp import convert_decibels_to_amplitude_ratio
+from ..utils.object_dict import ObjectDict
+
+
+class SpliceOut(BaseWaveformTransform):
+
+    """
+    spliceout augmentation proposed in https://arxiv.org/pdf/2110.00046.pdf
+    silence padding is added at the end to retain the audio length.
+    """
+
+    supported_modes = {"per_batch", "per_example"}
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        num_time_intervals=8,
+        max_width=400,
```

Comment:
I would still prefer to have this specified in seconds instead of samples, because this default is going to behave different based on the sample rate of the input. For example, 400 samples is 25 milliseconds with 16 kHz sample rate, but ~8 milliseconds with 48 kHz